src/shippingState.js

```diff
@@ -0,0 +1,271 @@
+    appState.setState({
+        selectedLocation: selectedLocation,
+    });
+
+    const detailedProviderBoxes = appState.getState().detailedProviderBoxes;
+    const detailedLocationBoxes = filterDetailBoxesByLocation(detailedProviderBoxes);
+
+    appState.setState({
+        detailedLocationBoxes: detailedLocationBoxes,
+    });
+}
+
+/**
+ * Update the state for allBoxesList and availableCollectionsObj.
+ * Remove the bags from the box.
+ * @param {string} boxId - the box with the bag to be removed.
+ * @param {array} bagsToMove - the bags to be removed from the box. 
+ */
+export const updateShippingStateRemoveBagFromBox = (boxId, bagsToMove) => {
+    addBagToAvailableCollections(boxId, bagsToMove);
+    removeBagFromBox(boxId, bagsToMove);
+}
+
+/**
+ * Update the state for allBoxesList and availableCollectionsObj.
+ * Add the bag to the box.
+ * @param {string} boxId - the boxId of the bag to be added
+ * @param {array} bagId - the bagId to be added to the box
+ * @param {object} boxToUpdate - the box object to be updated
+ */
+export const updateShippingStateAddBagToBox = (boxId, bagId, boxToUpdate) => {
+    addBagToBox(boxId, boxToUpdate);
+    removeBagFromAvailableCollections(bagId);
+}
+
+/**
+ * The bag is being removed from the box. Put it back in the availableCollectionsObj
+ * Build the bag and specimen data from detailedProviderBoxes data: availableCollectionsObj[bagId]: ['tube1Id', 'tube2Id', 'tube3Id']
+ * Isolate the last four digits of tubeId to match availableCollectionsObj format.
+ * Move object into availableCollectionsObj
+ * @param {*} boxId - the box with the bag being removed 
+ * @param {*} bagsToMove - the bags being removed from the box
+ */
+const addBagToAvailableCollections = (boxId, bagsToMove) => {
+    const availableCollectionsObj = appState.getState().availableCollectionsObj;
+    const detailedProviderBoxes = appState.getState().detailedProviderBoxes;
+
+    for (const bagId of bagsToMove) {
+        const collectionToMove = detailedProviderBoxes[boxId][bagId].arrElements;
+        const tubeIdArray = collectionToMove.map(tubeId => tubeId.slice(-4));
+
+        availableCollectionsObj[bagId] = tubeIdArray;
+    }
+
+    appState.setState({
+        availableCollectionsObj: {...availableCollectionsObj},
+    });
+}
+
+// Remove the bag from the box when user has clicked 'remove bag' in the 'View Shipping Box Contents' table.
+const removeBagFromBox = (boxId, bagsToMove) => {
+    const allBoxesList = appState.getState().allBoxesList;
+    const boxIndex = allBoxesList.findIndex(box => box[conceptIds.shippingBoxId] === boxId);
+
+    if (boxIndex !== -1) { 
+        for (const bagId of bagsToMove) {
+            delete allBoxesList[boxIndex].bags[bagId];
+        }
+
+        appState.setState({
+            allBoxesList: [...allBoxesList],
+        });
+    }
+}
+
+// Remove the bag from the availableCollectionsObj when it has been added to a box.
+const removeBagFromAvailableCollections = (bagId) => {
```

Comment:
If you want, the inside of function can be this:
```js
    appState.setState((state)=>{
      const availableCollectionsObj = state.availableCollectionsObj;
      delete availableCollectionsObj[bagId];
      return {availableCollectionsObj};
    });
```

Reply:
I like this, thanks for the example. I will leave it for now to match the other setState() operations stylistically.